terraform/alb.tf

```
@@ -0,0 +1,62 @@
+# --------------------------
+# ALB
+# --------------------------
+resource "aws_lb" "alb" {
+  name               = "${var.project}-alb"
+  internal           = false
+  load_balancer_type = "application"
+  security_groups = [
+    aws_security_group.alb_sg.id
+  ]
+  subnets = [for subnet in aws_subnet.public : subnet.id]
```

[Splat Expressions](https://developer.hashicorp.com/terraform/language/expressions/splat)で少し短く書けそう。

```suggestion
  subnets = aws_subnet.public[*].id
```

---

`aws_subnet.public[*].id`で実行するとエラーが出ました。
結論、values関数を使用することで解決できましたがエラーの原因について調べてみました。

***

### エラー文
```bash
╷
│ Error: Unsupported attribute
│ 
│   on ecs.tf line 42, in resource "aws_ecs_service" "tasting-note":
│   42:     subnets          = aws_subnet.ecs_private[*].id
│ 
│ This object does not have an attribute named "id".
╵
```

### outputで出力を調べてみた

#### `aws_subnet.public`
mapが返ってきました。
```bash
{
  "1a" = {
    arn = ....
  }
  "1c" = {
    arn = ...
  }
}
```

#### `aws_subnet.public[*]`
map型の要素を持つlistが返ってきました。
```bash
[
  {
    "1a" = {
      arn = ....
    }
    "1c" = {
      arn = ...
    }
  }
]
```
これに対してidにアクセスしていたためエラーが発生していました!!

### まとめ
`aws_subnet.public`をvalues関数で囲むことで`id`を含むオブジェクトの配列が返ってくるのでそれに対して`[*].id`とすることで各サブネットのidを要素に持つ配列を取得できました！

---

@yuma-matsui 
すみません、これは私のレビューがダメでした...
docに `for expressions` を使うように書かれていたので、元のコードで良かったようです。

> Splat Expressions with Maps
The splat expression patterns shown above apply only to lists, sets, and tuples. To get a similar result with a map or object value you must use `for expressions`.
> https://developer.hashicorp.com/terraform/language/expressions/splat#splat-expressions-with-maps

修正コミットを force-push で消すか、revert してしまった方が良いかも。